src/services/item/plugins/app/appSetting/repository.ts

```diff
@@ -46,16 +50,21 @@ export const AppSettingRepository = AppDataSource.getRepository(AppSetting).exte
     return appSettingId;
   },
 
-  async get(id: string) {
+  async get(id: string): Promise<AppSetting> {
     // additional check that id is not null
     // o/w empty parameter to findOneBy return the first entry
     if (!id) {
       throw new AppSettingNotFound(id);
     }
-    return this.findOneBy({ id });
+    return this.findOne({ where: { id }, relations: { creator: true, item: true } });
   },
 
-  getForItem(itemId: string) {
-    return this.findBy({ item: { id: itemId } });
+  getForItem(itemId: string): Promise<AppSetting[]> {
+    return this.find({
```

Comment:
To avoid any error, we check that `itemId` really exists (even if it's not set as optional here). Otherwise `this.find` will return everything which would just explode. If you could add this check!